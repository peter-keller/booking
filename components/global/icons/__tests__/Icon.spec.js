import { shallowMount } from '@vue/test-utils'

import IconSize from '../size'
import Icon from '..'

describe('Elements: Icon', () => {
  const defaultOpts = {
    propsData: {
      name: 'Hello'
    }
  }

  const factory = (opts = {}) => shallowMount(Icon, {
    ...defaultOpts,
    ...opts
  })

  describe('snapshots', () => {
    it('matches default snapshot', () => {
      const wrapper = factory()

      expect(wrapper.html()).toMatchSnapshot()
    })

    Object.values(IconSize).forEach(size => {
      it(`matches snapshot when size === ${size}`, () => {
        const wrapper = factory({ propsData: { ...defaultOpts.propsData, size } })

        expect(wrapper.html()).toMatchSnapshot()
      })
    })
  })

  it('$emits click on click if it is clickable', () => {
    const wrapper = factory({
      propsData: { ...defaultOpts.propsData, clickable: true }
    })

    wrapper.trigger('click')

    expect(wrapper.emitted('click')).toBeTruthy()
  })

  it('does not $emits click on click if it is not clickable', () => {
    const wrapper = factory({
      propsData: {
        name: 'Hello',
        clickable: false
      }
    })

    wrapper.trigger('click')

    expect(wrapper.emitted('click')).toBeFalsy()
  })

  describe('#Svg', () => {
    it('returns empty string when no icons match', () => {
      const wrapper = factory({
        propsData: {
          name: 'asasd'
        }
      })

      expect(wrapper.vm.svg).toBe('')
    })
  })
})
