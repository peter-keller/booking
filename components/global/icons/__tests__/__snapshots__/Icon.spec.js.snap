// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements: Icon snapshots matches default snapshot 1`] = `"<div class=\\"iconWrapper iconWrapper--medium\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === large 1`] = `"<div class=\\"iconWrapper iconWrapper--large\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === medium 1`] = `"<div class=\\"iconWrapper iconWrapper--medium\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === small 1`] = `"<div class=\\"iconWrapper iconWrapper--small\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === xlarge 1`] = `"<div class=\\"iconWrapper iconWrapper--xlarge\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === xsmall 1`] = `"<div class=\\"iconWrapper iconWrapper--xsmall\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === xxlarge 1`] = `"<div class=\\"iconWrapper iconWrapper--xxlarge\\"></div>"`;

exports[`Elements: Icon snapshots matches snapshot when size === xxsmall 1`] = `"<div class=\\"iconWrapper iconWrapper--xxsmall\\"></div>"`;
